picamera2/previews/qt_previews.py

```diff
@@ -105,8 +105,6 @@ def stop(self):
             retq = Queue()
             QtPreviewBase.previewcreateq.put((Command.DELETE, retq, (self, self.qpicamera2)))
             retq.get()
-            del self.qpicamera2
```

Comment:
Just wondering why this part was removed? I thought this was necessary to remove 
the preview when stop was called.

Reply:
Yes, good question. I wish I had a convincing answer...!

I decided to make it the widget's job to do all the cleaning up, so once you get here after the `Command.DELETE` has happened, `self.qpicamera2` is already `None`. So I would have to store `self.qpicamera2` beforehand, and then delete it after. Or maybe the `qpicamera2`'s `cleanup` method would delete itself? It all felt a bit weird.

So anyway, I wondered whether I really needed it if I have dropped all the handles to it... and it _seems_ like I don't. At least, all the tests pass, and we did go to some effort to test opening/closing things. But I'm well aware that this isn't conclusive. All in all I'm a bit nervous of all this Qt stuff, so more testing is certainly warranted...